src/win/fs.c

```diff
@@ -627,6 +644,10 @@ void fs__write(uv_fs_t* req) {
   if (offset != -1) {
     memset(&overlapped, 0, sizeof overlapped);
     overlapped_ptr = &overlapped;
+    if (SetFilePointerEx(handle, zero_offset, &original_position,
+                         FILE_CURRENT)) {
+      restore_position = 1;
```

Comment:
On which cases can this fail? Should we note in the documentation that on Windows the offset could actually be modified after calling `uv_fs_write() / uv_fs_read()`?